src/runtime-rs/crates/hypervisor/src/dragonball/inner.rs

```diff
@@ -188,8 +189,12 @@ impl DragonballInner {
 
     fn set_vm_base_config(&mut self) -> Result<()> {
         let serial_path = [&self.run_dir, "console.sock"].join("/");
-        let (mem_type, mem_file_path) = if self.config.memory_info.enable_hugepages {
-            (String::from(HUGETLBFS), String::from(DEV_HUGEPAGES))
+        let (mem_type, mem_file_path) = if !self.config.memory_info.hugepages.is_empty() {
+            match self.config.memory_info.hugepages.as_str() {
+                HUGE_PAGE_MODE_THP => (String::from(HUGE_SHMEM), String::from("")),
```

Comment:
Besides `hugeshmem`, Dragonball could also use `hugeanon`, `hugemmap` and `hugememfd` to trigger transparent huge page.

Reply:
![image](https://user-images.githubusercontent.com/31121654/234349503-45735c79-010e-466f-b15c-f7d1c4931499.png)

Reply:
THP couldn't represent all of these types. Should we expose all of these memory backend to runtime?

Reply:
make sense. Then the current code is good for me.